client/src/components/Calendar.jsx

```diff
@@ -7,46 +7,139 @@ class Calendar extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      currentDate: moment().format('L'),
       monthL0: moment().subtract(1, 'M').format('MMMM'),
+      dateL0: moment().subtract(1, 'M').format('L'),
       monthL1: moment().format('MMMM'),
+      dateL1: moment().format('L'),
       monthR1: moment().add(1, 'M').format('MMMM'),
-      monthR2: moment().add(2, 'M').format('MMMM')
+      dateR1: moment().add(1, 'M').format('L'),
+      monthR2: moment().add(2, 'M').format('MMMM'),
+      dateR2: moment().add(2, 'M').format('L')
     };
+
+    this.previousMonths = this.previousMonths.bind(this);
+    this.nextMonths = this.nextMonths.bind(this);
+  }
+
+
+  previousMonths() {
+    var previous, current, next, afterNext;
+    var date = this.state.currentDate;
+
+    current = this.subtractMonth(date);
+    next = date;
+    afterNext = this.addMonth(date);
+    previous = this.subtractMonth(current);
+
+    this.setState({
+      currentDate: current,
```

Comment:
since previous and nextMonth use the same object for setting state you could abstract that out to a variable and save yourself some repetition I think

Reply:
Nice catch!